app/controllers/concerns/recovery_code_concern.rb

```diff
@@ -0,0 +1,12 @@
+module RecoveryCodeConcern
+  delegate :active_profile, to: :current_user
+
+  def create_new_code
+    if active_profile.present?
+      Pii::ReEncryptor.new(current_user, user_session).perform
+      active_profile.recovery_code
```

Comment:
I don't see the naming thing as a snag. `Profile.encrypt_pii` and `Profile.encrypt_recovery_pii` already return the recovery code.